modules/db/src/main/scala/gem/dao/TargetDao.scala

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) 2016-2017 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package gem
+package dao
+
+import doobie._
+import doobie.implicits._
+import gem.dao.meta._
+import gem.dao.composite._
+import gem.enum.TrackType
+import shapeless._
```

Comment:
Are we crossing the Rubicon here, or maybe we already did and I didn't notice.  @cquiroz any thoughts?  I guess I'd like to know whether we're going to try to only use certain parts, or only use it in certain sub-projects, or maybe we're ready to fully commit?  It seems a little less frightening after a brief intro at Scala World but it seems like one of those things that if it spreads very far there's no going back.

Reply:
I totally agree. I'd like that we somehow get all up to speed on shapeless before going too far. I guess it also depends a lot on what parts we use. Just using `HList` in same places is one thing, impilict recursive typeclass derivation is quite another, and there are far more advanced (esoteric) features than that 😄

Reply:
@tpolecat could you briefly explain what you need shapless here for? I admit I haven't read the PR, possibly it is quite obvious

Reply:
On line 48 `enc.composite` has type `Composite[Sidereal :+: Nonsidereal :+: CNil]` and shapeless is needed here to provide the implicits required for the call to `unify` which extracts the value as a `Track` (which is the LUB of `Sidereal` and `Nonsidereal`).

I could probably provide this as a method on `TaggedCoproductEncoder` so we don't need shapeless at the call site, if you'd prefer to keep it under wraps.

Reply:
> I could probably provide this as a method on TaggedCoproductEncoder so we don't need shapeless at the call site, if you'd prefer to keep it under wraps.

That would probably be good, at least until we're all confident and competent with Shapeless.

Reply:
Ok I'll give that a try.